### DISCUSSION TITLE
Use BitArray in ObservationState

### DISCUSSION THREAD

**Discussion by tkunst891** [pointfreeco/swift-composable-architecture#2737] [Q&A]
I have a feature which uses a BitArray (from Swift Collections) as a property in its State. How could I extend BitArray to make the individual bits observable? Currently change of any bit is updating all views which observe this property, even if the views access only one bit.

**Comment by mbrandonw** [comment]
Hi @tkunst891, it most likely is a matter of implementing an overload of `_$isIdentityEqual` that is tuned specifically for bit arrays. We can see all the overloads we have [here](https://github.com/pointfreeco/swift-composable-architecture/blob/1e7942c9d182b808d8260eb1840f01d51590df02/Sources/ComposableArchitecture/Observation/ObservableState.swift#L104).

If that doesn't help then perhaps you can provide us with a simple project that reproduces the problem you are seeing. I'm not super familiar with bit arrays, and so would need a project to play around with to see what is going on.
